table/queries/merged_reduced_scans.sql

```diff
@@ -26,6 +26,170 @@ CREATE TEMP FUNCTION AddOutcomeEmoji(outcome STRING) AS (
+  resolver_connect_error_rate FLOAT64,
+  # Failure to get a valid certificate when fetching the page from a returned IP.
+  resolver_invalid_cert_rate FLOAT64,
+  resolver_non_zero_rcode_rate FLOAT64
+) AS (
+  resolver_connect_error_rate > .75 OR
+  resolver_invalid_cert_rate > .75 OR
+  resolver_non_zero_rcode_rate > .75 
+);
+
+CREATE TEMP FUNCTION SatelliteOutcomeString(domain_name STRING,
+                                            dns_error STRING,
+                                            rcode INTEGER,
+                                            answers ANY TYPE) AS (
+    CASE 
+        WHEN (dns_error is NOT NULL
+              AND dns_error != ""
+              AND dns_error != "null"
+              AND dns_error != "SERVFAIL") THEN ClassifySatelliteError(dns_error)
+        # TODO fix -1 rcodes in v1 data in the pipeline
+        WHEN rcode = -1 THEN "❓read/udp.timeout"
+        WHEN rcode != 0 THEN ClassifySatelliteRCode(rcode)
+        WHEN ARRAY_LENGTH(answers) = 0 THEN "❗️answer:no_answer"
+        ELSE IFNULL(
+            (SELECT InvalidIpType(answer.ip) FROM UNNEST(answers) answer LIMIT 1),
+            CASE
+                WHEN (SELECT LOGICAL_OR(answer.matches_control.ip)
+                      FROM UNNEST(answers) answer)
+                      THEN "✅answer:matches_ip"
+                WHEN (SELECT LOGICAL_OR(a.https_tls_cert_matches_domain AND a.https_tls_cert_has_trusted_ca)
+                      FROM UNNEST(answers) a)
+                      THEN "✅answer:valid_cert"
+                WHEN (SELECT LOGICAL_OR(a.https_tls_cert_matches_domain AND NOT a.https_tls_cert_has_trusted_ca)
+                      FROM UNNEST(answers) a)
+                      THEN CONCAT("❗️answer:invalid_ca_valid_domain:", answers[OFFSET(0)].https_tls_cert_issuer)
+                WHEN (SELECT LOGICAL_AND(NOT a.https_tls_cert_matches_domain)
+                      FROM UNNEST(answers) a)
+                      THEN CONCAT("❗️answer:cert_not_for_domain:", answers[OFFSET(0)].https_tls_cert_common_name)
+                WHEN (SELECT LOGICAL_OR(answer.http_analysis_is_known_blockpage)
+                      FROM UNNEST(answers) answer)
+                      THEN CONCAT("❗️page:http_blockpage:", answers[OFFSET(0)].http_analysis_page_signature)
+                WHEN (SELECT LOGICAL_OR(answer.https_analysis_is_known_blockpage)
+                      FROM UNNEST(answers) answer)
+                      THEN CONCAT("❗️page:https_blockpage:", answers[OFFSET(0)].https_analysis_page_signature)
+                -- We check AS after cert/blockpage because we've seen (rare) cases of blockpages hosted on the ISP that also hosts Akamai servers.
+                WHEN (SELECT LOGICAL_OR(answer.matches_control.asn)
+                      FROM UNNEST(answers) answer)
+                      THEN "✅answer:matches_asn"
+                ELSE CONCAT("❓answer:not_validated:", AnswersSignature(answers))
```

Comment:
TODO: rename to "fetch failed" or something clearer. This indicates we were not able to establish a TLS connection which is clearly unexpected.

Reply:
this was worked on seperatly in https://github.com/censoredplanet/censoredplanet-analysis/pull/247